### PR TITLE
update error codes

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

File: sign/error-codes.md [hunk 1]
```diff
@@ -29,5 +30,11 @@ case .invalidMethod: return 1001
...
 
 case .invalidUpdateRequest: return 1003
-case .invalidUpdateExtendRequest: return 1004
+case .invalidExtendRequest: return 1004
+```
+
+## REMAINING
+
+```sh
+case .userDisconnected: return 6000
```
Comment (jakubuid) [inline sign/error-codes.md:41]: Haven't we agreed to add more disconnection reasons as well?

File: sign/error-codes.md [hunk 2]
```diff
@@ -29,5 +30,12 @@ case .invalidMethod: return 1001
...
+case .sessionSettlementFailed: return 6001
```
Comment (Elyniss) [inline sign/error-codes.md]: In Kotlin SDK there is a chance that the session cannot be settled due to a database error. Then return this error code.

**Comment by Elyniss** [comment]
Complemented test cases with their error codes

**Review by TalhaAli00 — APPROVED** [review]

**Review by IljaDaderko — APPROVED** [review]

File: sign/error-codes.md (hunk 2 above)
Comment (llbartekll) [reply sign/error-codes.md]: 6000 and 6001 are not much related so maybe for `sessionSettlementFailed` you should give 7000 or something else?
another thing is that `userDisconnected` is actually not an error but more like a reason code. it is included as a param in a request in contrast to all others that are included in error responses.

so should we call the doc `error-codes.md`?

Comment (Elyniss) [reply sign/error-codes.md]: I see what you mean with `userDisconnected`. I've updated the specs. What do you think of them now?

Comment (llbartekll) [reply sign/error-codes.md]: `protocol-codes.md` ?
now I like it more.

**Review by llbartekll — APPROVED** [review]

File: sign/error-codes.md (hunk 2 above)
Comment (Elyniss) [reply sign/error-codes.md]: I would rather stick with `codes.md` rather than `protocol-codes.md`. IMO, sounds/reads better.

**Pull request merged**